tools/windows-node-installer/cmd/azure.go

```diff
@@ -0,0 +1,144 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/openshift/windows-machine-config-operator/tools/windows-node-installer/pkg/cloudprovider"
+	"github.com/openshift/windows-machine-config-operator/tools/windows-node-installer/pkg/cloudprovider/azure"
+	"github.com/spf13/cobra"
+)
+
+// azCreateFlagInfo contains azure specific information for creating an instance.
+// the fields inside the struct gets filled once the flags are parsed.
+// all the fields are optional except credentialPath.
+var azCreateFlagInfo struct {
+	imageID             string
+	instanceType        string
+	credentialPath      string
+	credentialAccountID string
+	// Provide the IP address if the installer doesn't want to create one.
+	ipName string
+	// Provide the nic name if the installer doesn't want to create one.
+	nicName string
+}
+
+func init() {
+	azureCmd := newAZCmd()
+	rootCmd.AddCommand(azureCmd)
+	azureCmd.AddCommand(azCreateCmd())
+	azureCmd.AddCommand(azDestroyCmd())
+}
+
+func newAZCmd() *cobra.Command {
+	azureCmd := &cobra.Command{
+		Use:   "azure",
+		Short: "Takes azure specific resource names from user",
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			return requiredAZFlags(cmd)
+		},
+	}
+	azureCmd.PersistentFlags().StringVar(&azCreateFlagInfo.credentialPath, "credentials", "",
+		"file location to the azure cloud provider credentials (required).")
+
+	return azureCmd
+}
+
+// requiredazureFlags defines required flags for createInfo.
+func requiredAZFlags(azureCmd *cobra.Command) error {
+	err := azureCmd.MarkPersistentFlagRequired("credentials")
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// createCmd defines `create` command and creates a Windows instance using parameters from the persistent flags to
+// fill up information in azCreateFlagInfo.
+func azCreateCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create a Windows node on the Azure cloud provider.",
+		Long: "creates a Windows instance under the same Vnet of" +
+			"the existing OpenShift cluster. " +
+			"The created instance would be used as a worker node for the OpenShift Cluster.",
+		TraverseChildren: true,
+		RunE: func(_ *cobra.Command, _ []string) error {
+			os.Setenv("AZURE_AUTH_LOCATION", azCreateFlagInfo.credentialPath)
```

Comment:
right it will throw an error if the env variable AZURE_AUTH_LOCATION is set to empty string.
Basically something like this Error: unable to get the settings from path: environment variable AZURE_AUTH_LOCATION is not set